modnet/sklearn.py

```diff
@@ -0,0 +1,134 @@
+__author__ = "David Waroquiers, Matgenix SRL"
+__credits__ = "David Waroquiers, Matgenix SRL"
+"""Proposition of API for MODNet.
+
+
+The idea is to do something like :
+from sklearn.pipeline import Pipeline
+modnet_featurizer = MODNetFeaturizer(...arguments here...)
+rr_analysis = RR(...arguments here...)
+modnet_model = MODNet(...arguments here...)
+p = Pipeline([('featurizer', modnet_featurizer), ('rr', rr_analysis), ('modnet', modnet_model)])
+
+* One note about scikit learn's steps when performing cross-validation:
+  A given transformer (e.g. PCA, or RR) will not be executed at each step of the cross validation if you cache it, i.e.
+  scikit-learn detects that the inputs are the same (if they are indeed the same) and does not do it several times
+  (see https://scikit-learn.org/stable/modules/compose.html#caching-transformers-avoid-repeated-computation).
+* Another note about the fit method of a Pipeline object:
+  It is possible to pass fit parameters to each step of a Pipeline object
+  (see https://scikit-learn.org/stable/modules/generated/sklearn.pipeline.Pipeline.html#sklearn.pipeline.Pipeline.fit)
+* Another note about the __init__ method of sklearn derived estimators:
+  The method should ONLY set the instance variables. In particular, "every keyword argument accepted by __init__
+  should correspond to an attribute on the instance" (as stated in sklearn's developer documentation)
+  (see https://scikit-learn.org/stable/developers/develop.html#instantiation).
+"""
+
+
+from sklearn.base import BaseEstimator
+from sklearn.base import RegressorMixin
+from sklearn.base import TransformerMixin
+from typing import Union
+from typing import Dict
+
+
+class MODNetFeaturizer(TransformerMixin, BaseEstimator):
+    """"""
+
+    def __init__(self, matminer_featurizers):
+        """Constructor for MODNetFeaturizer"""
+        self.matminer_featurizers = matminer_featurizers
+
+    def fit(self, X, y=None):
```

Comment:
What should be X ? Structure most probably, but what if one only has the composition?

Reply:
I would say X should be a pandas DataFrame containing at least "structure" or "composition" as a column. Then of course if you pass featurizers that require the structure but you only have the composition, the code should raise an error